homework3/hw3/ViewModel/MenuViewModel.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  MenuViewModel.swift
+//  hw3
+//
+//  Created by Tina Jureško on 19.03.2025..
+//
+
+import Foundation
+import UIKit
+import SofaAcademic
+import SnapKit
+
+
+struct MenuViewModel {
+    var selectedSport: SportType
+    var onSportSelectionChanged: ((SportType) -> Void)?
+    
+    init(selectedSport: SportType) {
+        self.selectedSport = selectedSport
+    }
+    
+    mutating func selectSport(_ sport: SportType) {
+        self.selectedSport = sport
+        
+        onSportSelectionChanged?(sport)
+    }
+    
+    func getSportCell(for sport: SportType) -> UIView? {
```

Comment:
Idealno bi bilo da uopće u ViewMOdelu nemamo import UIKit i ništa vezano za View. Trebalo bi samo exposati state koji nam treba npr. za UIColor ili UIIMage, iako ajde za te 2 vrijednosti još može proći ViewModel.

Ali ViewModel po pesu bi bio samo da imamo import Foundation i to je to. 

Mislim da ti je ovaj View dosta basic da imaš ViewModel za njega.
Ovo Da dohvaćamo neki UIView je dosta probijanje patterna, to treba biti u Viewu. z